src/float.rkt

```diff
@@ -56,14 +56,28 @@
         [bf->repr (representation-bf->repr repr)])
     (bfinfinite? (repr->bf (bf->repr +inf.bf)))))
 
+;; If the representation uses saturation instead of overflow
+(define (probably-saturated? ordinal repr)
+  (define bfval (ordinal->bigfloat ordinal))
+  (define bfval2 (bf* bfval 2.bf))
+  (define ->repr (representation-bf->repr repr))
+  (define ->ordinal (representation-repr->ordinal repr))
+  (= (->ordinal (->repr bfval))
+     (->ordinal (->repr bfval2))))
+
 (define (bound-ordinary-values repr)
-  (if (has-infinite-value? repr)
+  (if (has-infinite-value? repr) ; bail if representation does not have +inf
       (parameterize ([bf-rounding-mode 'nearest])
-        (let loop ([ordinal (bigfloat->ordinal (largest-ordinary-value repr))] [stepsize 1])
-          (define bfval (ordinal->bigfloat ordinal))
-          (if (bfinfinite? ((representation-repr->bf repr) ((representation-bf->repr repr) bfval)))
-              bfval
-              (loop (+ ordinal stepsize) (* stepsize 2)))))
+        (define hi-ordinal (bigfloat->ordinal (largest-ordinary-value repr)))
+        (if (probably-saturated? hi-ordinal repr) ; bail if representation uses saturation
+            (ordinal->bigfloat hi-ordinal)
+            (let loop ([ordinal hi-ordinal] [stepsize 1])
+              (define bfval (ordinal->bigfloat ordinal))
+              (define ->repr (representation-bf->repr repr))
+              (define ->bf (representation-repr->bf repr))
+              (if (bfinfinite? (->bf (->repr bfval)))
+                  bfval
+                  (loop (+ ordinal stepsize) (* stepsize 2))))))
```

Comment:
Not sure I understand what's going on here. Is it a separate bug?

Reply:
This procedure returns the largest bigfloat that still rounds to a finite value in the representation. This will break for posits since they use saturation. Thus, if we detect saturation (not sure how sound my implementation is), we should just bail rather than continually incrementing.

Reply:
Right—I'm asking what it has to do with splitting the operator table. Is this an unrelated issue?

Reply:
Oh got it. Yes unrelated. It appeared when I updated softposit-herbie to inherit interval operations